### DISCUSSION TITLE
FairSoft Installation Taking Lots Of Time.

### DISCUSSION THREAD

**Discussion by rishavpandey160999** [FairRootGroup/FairSoft#475] [Q&A]
Hi, 

I'm trying to install FiairSoft in my PC. It's taking a lot of time. From 0-50% it took around 2 hours, and now it seems to be almost freezed. For building an individual package, like "geant4"  and "root", it's taking around 2 hours for each. I'm trying to install the same in my virtual machine of 3GB RAM. So, can anyone tell me how long it takes to install Fairsoft, with this architecture?

Thankyou in advance!

**Comment by bus786** [comment]
i tried to installation of cbmroot both method first auto- installation and second is a manual installation in UBUNTU 22.04 system and it has 8gb space but i am unable to do that some time system is hanged or terminal is closed after completing 43% of fairsoft package and some time terminal is closed . so is there any offical statement for cbmroot we need at least 16 gb or not ? for me cbmroot is newthing so guide me please.

**Reply by karabowi** [reply]
The full FairSoft+FairRoot+CbmRoot = 23GB + 1GB + 7GB will take a bit over 30GB space. After FairSoft installation one can remove build/Source to free around 4GB. The whole process on a typical machine should take 6+ hours.

**Comment by bus786** [comment]
Dear Sir ,
I have a 30 GB Space in my desktop, so can you please tell me what is the
criteria for installation of FAIRSOFT , FAIRROOT, CBMROOT.
Regards,
Bushra Parveen

**Comment by bus786** [comment]
Dear Sir ,
I have a 30 GB Space in my desktop, so can you please tell me what is the
criteria MEMORY for installation of FAIRSOFT , FAIRROOT, CBMROOT.
Regards,
Bushra Parveen

**Reply by karabowi** [reply]
I do not think we have any specific requirements about memory. We had no complaints recently.
IMHO it is more probable that any experiment (like CBM) requires much more memory
for the simulation/analysis than we for the software installation.
Having said that, I think that 8GB is reasonable minimum, but installation should run even on less RAM.

**Comment by bus786** [comment]
Actually Sir  I try many times fairsoft installation it reach upto 33% or
43% after that terminal is automatic closed or system is hanged and no
error show .  That's why i ask about memory quantity needed for fairsoft .
In my system UBUNTU 22.04 is installed . so sir please guide me.

**Comment by karabowi** [comment]
Can you please go to your FairSoft build/Log directory and to "ls -ltrh"
there? As the last few files, you should see the logs of the package 
that caused trouble. Please send them over.

By the way, did you run the `FairSoft/legacy/setup-ubuntu.sh` script before the installation?

Which version of `FairSoft` are you trying to install?
